src/test/template-shadowroot_test.ts

```diff
@@ -4,12 +4,40 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 
-import {hydrateShadowRoots as manualWalkHydration} from '../_implementation/manual_walk.js';
-import {transformShadowRoots} from '../_implementation/mutation_observer';
-import {hydrateShadowRoots as querySelectorHydration} from '../_implementation/queryselectorall.js';
-import {hasNativeDeclarativeShadowRoots} from '../template-shadowroot.js';
+import { hydrateShadowRoots as manualWalkHydration } from '../_implementation/manual_walk.js';
+import { transformShadowRoots } from '../_implementation/mutation_observer';
+import { hydrateShadowRoots as querySelectorHydration } from '../_implementation/queryselectorall.js';
+import { hasNativeDeclarativeShadowRoots } from '../template-shadowroot.js';
+import { expect } from '@esm-bundle/chai';
 
-const elementLog: Array<string|null> = [];
+const elementLog: Array<string | null> = [];
+
+// lib.dom.ts is out of date, so declare our own parseFromString here.
+interface DOMParser {
+  parseFromString(
+    string: string,
+    type: DOMParserSupportedType,
+    options?: {
+      includeShadowRoots: boolean;
+    }
+  ): Document;
+}
+
+function renderFragment(html: string) {
+  const fragment = (new DOMParser() as DOMParser).parseFromString(
+    html,
+    'text/html',
+    {
+      includeShadowRoots: true,
+    }
+  );
+
+  const root = document.createElement('div');
+  root.appendChild(fragment.body.firstChild!);
+  document.body.append(root);
+
+  return root;
+}
```

Comment:
Shouldn't we specifically avoid this and set `.innerHTML` since we want to avoid the browser parsing and creating the shadowroot from the template element?

Reply:
Spoke in person: We need to do this because innerHTML will not auto-attach the shadowroots in browsers that support DSD. The problem is that due to a previous PR, it seems that the polyfill will no longer run if DSD is supported. That means these tests would fail in browsers that support DSD because the ponyfill will not attempt to upgrade the templates passed in via innerHTML

Reply:
Removing the check for native DSD should probably be considered in a future PR though